Mesh_3/test/Mesh_3/test_c3t3_io.cpp

```diff
@@ -446,19 +481,135 @@ struct Test_c3t3_io {
+    static Corner_index get_corner_index_2() { return 8; }
+
+    static std::string reference_format_string()
+    {
+      return "Triangulation_3(Weighted_point<Point_3>,Vb(Tvb_3+i+i),Cb(i+RTcb_3+(i)[4]))";
+    }
+  };
+
+  typedef CGAL::Mesh_triangulation_3<MD_1, K>::Type Tr_1;
+  typedef MD_homogeneous_types MD_2;
+  typedef MD_1::Curve_index Curve_index_1;
+  typedef MD_1::Corner_index Corner_index_1;
+  typedef  Tr_1::Point Point;
+  typedef  Tr_1::Vertex_handle Vertex_handle;
+  typedef  Tr_1::Edge Edge;
+  typedef  Tr_1::Facet Facet;
+  typedef  Tr_1::Cell_handle Cell_handle;
+
+  typedef CGAL::Mesh_complex_3_in_triangulation_3<
+    Tr_1,
+    Corner_index_1,
+    Curve_index_1
+    > C3t3_1;
+
+  typedef MD_2::Curve_index Curve_index_2;
+  typedef MD_2::Corner_index Corner_index_2;
+
+  typedef CGAL::Mesh_triangulation_3<MD_2, K>::Type Tr_2;
+  typedef CGAL::Mesh_complex_3_in_triangulation_3<
+    Tr_2,
+    Corner_index_2,
+    Curve_index_2
+    > C3t3_2;
+
+
+  C3t3_1 c3t3;
+  C3t3_2 c3t3_2;
+  Tr_1& tr = c3t3.triangulation();
+  Vertex_handle v1 = tr.insert(Point(10,11,12));
+  Vertex_handle v2 = tr.insert(Point(11,13,10));
+  Vertex_handle v3 = tr.insert(Point(7,4,6));
+  Vertex_handle v4 = tr.insert(Point(5,2,14));
+  Vertex_handle v5 = tr.insert(Point(1,2,3));
+  Vertex_handle v6 = tr.insert(Point(3,9,13));
+
+  Edge e1 = *(c3t3.triangulation().finite_edges_begin());
+  Edge e2 = *(++c3t3.triangulation().finite_edges_begin());
+
+  Facet f1 = *(c3t3.triangulation().finite_facets_begin());
+  Facet f2 = *(++c3t3.triangulation().finite_facets_begin());
+
+  Cell_handle c1 = c3t3.triangulation().finite_cells_begin();
+  Cell_handle c2 = ++c3t3.triangulation().finite_cells_begin();
+
+  c3t3.add_to_complex(c2, MD_1::get_sub_domain_index_2());
+  c3t3.add_to_complex(c1, MD_1::get_sub_domain_index_1());
+  c3t3.add_to_complex(f1, MD_1::get_surface_patch_index_1());
+  c3t3.add_to_complex(f2, MD_1::get_surface_patch_index_2());
+  c3t3.add_to_complex(e1, MD_1::get_curve_index_1());
+  c3t3.add_to_complex(e2, MD_1::get_curve_index_2());
+  c3t3.add_to_complex(v1, MD_1::get_corner_index_1());
+  c3t3.add_to_complex(v2, MD_1::get_corner_index_2());
+
+  // Fill indices in various faces (with different dimensions)
+  v1->set_dimension(0);
+  v1->set_index(MD_1::get_corner_index_1());
+  v2->set_dimension(0);
+  v2->set_index(MD_1::get_corner_index_2());
+  v3->set_dimension(1);
+  v3->set_index(MD_1::get_curve_index_1());
+  v4->set_dimension(1);
+  v4->set_index(MD_1::get_curve_index_2());
+  v5->set_dimension(2);
+  v5->set_index(MD_1::get_surface_patch_index_1());
+  v6->set_dimension(3);
+  v6->set_index(MD_1::get_sub_domain_index_1());
+
+  std::ios_base::openmode mode = std::ios_base::in | std::ios_base::out;
+  std::stringstream stream(mode);
+  CGAL::Mesh_3::save_binary_file(stream, c3t3, false);
+  assert(stream);
+  CGAL::Mesh_3::load_binary_file(stream, c3t3_2);
+  assert(stream);
+    std::cout << "Content of c3t3_1:\n"
```

Comment:
@lrineau dummy question: can we dump both c3t3 in `std::string` and assert that the strings are equal?

Reply:
IIRC it is what we do in the determinism test (but with the ASCII version, not with the binary version)

Reply:
So we can update the test and test this branch, right?

Reply:
Yes, I think so.

Reply:
@janetournois would you mind updating the test?

Reply:
Ok I'm doing it